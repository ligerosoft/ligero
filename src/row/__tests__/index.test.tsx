import { mount, render } from 'enzyme';
import React from 'react';
import Row from '..';

describe('Row', () => {
  it('mount row correctly', () => {
    expect(mount(<Row />)).toMatchSnapshot();
  });

  it('should has "ligero-row--wrap" default or set wrap props is true', () => {
    const wrapper = render(<Row />);
    expect(wrapper.hasClass('ligero-row--wrap')).toBe(true);
  });
  it('render classes when set props align or justify', () => {
    const wrapper = mount(<Row />);
    wrapper.setProps({ align: 'center' });
    const aligns = ['top', 'center', 'bottom'];

    aligns.forEach(align => {
      wrapper.setProps({ align });
      expect(wrapper.render().hasClass(`ligero-row--align-${align}`)).toBe(true);
    });
    const justifies = ['start', 'center', 'end', 'space-around', 'space-between'];

    justifies.forEach(justify => {
      wrapper.setProps({ justify });
      expect(wrapper.render().hasClass(`ligero-row--justify-${justify}`)).toBe(true);
    });
  });
});
