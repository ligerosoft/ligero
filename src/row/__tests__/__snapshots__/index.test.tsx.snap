// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row mount row correctly 1`] = `
<@ligero/row>
  <div
    className="ligero-row ligero-row--wrap"
  />
</@ligero/row>
`;
